### PR TITLE
zib-AdministrationAgreement

### PR DESCRIPTION
**Pull request by ArdonToonstra**

### CODE REVIEW

File: resources/zib/zib-AdministrationAgreement.xml [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+  <contact>
+    <name value="Nictiz" />
+    <telecom>
+      <system value="email" />
+      <value value="info@nictiz.nl" />
+      <use value="work" />
+    </telecom>
+  </contact>
+  <description value="An administration agreement is the use (or administering) instructions from the pharmacist to the patient (or their representative or administrator), whereby a medication agreement is structured at a concrete level." />
+  <purpose value="This MedicationDispense resource represents the Dutch [zib ('Zorginformatiebouwsteen', i.e. Health and Care Information Model) ](https://zibs.nl/wiki/AdministrationAgreement-v1.0.3(2020EN))." />
```
Comment (Ashwin-nictiz) [inline resources/zib/zib-AdministrationAgreement.xml:18]: Not confirm the guidelines missing the English zib name 

This [resource type] resource represents the Dutch [zib ('Zorginformatiebouwsteen', i.e. Health and Care Information Model) [English zib name] [version]([release])]([link to the English zib page on zibs.nl]).

File: resources/zib/zib-AdministrationAgreement.xml [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+      <path value="MedicationDispense.status" />
+      <short value="CanceledIndicator" />
+      <definition value="A code specifying the state of the set of dispense events. In the event of an error correction, the value of this element shall be 'entered-in-error'." />
+      <alias value="GeannuleerdIndicator" />
+      <mapping>
+        <identity value="zib-administrationagreement-v1.0.3-2020EN" />
+        <map value="NL-CM:9.8.23034" />
+        <comment value="CanceledIndicator (if status = 'entered-in-error' then CanceledIndicator = true)" />
+      </mapping>
+    </element>
```
Comment (Ashwin-nictiz) [inline resources/zib/zib-AdministrationAgreement.xml:174]: The definition is not right the zib says:

"In the event of an error correction, this indicator is to be put on for the incorrect agreement"

Also there is still a ongoing discussion about the cardinality 

https://github.com/Nictiz/Nictiz-R4-zib2020/issues/97

File: resources/zib/zib-AdministrationAgreement.xml [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+      <short value="Comment" />
+      <definition value="Comments on the administration agreement." />
+      <alias value="Toelichting" />
+      <max value="1" />
+      <mapping>
+        <identity value="zib-administrationagreement-v1.0.3-2020EN" />
+        <map value="NL-CM:9.8.22275" />
+        <comment value="Comment" />
+      </mapping>
+    </element>
```
Comment (Ashwin-nictiz) [inline resources/zib/zib-AdministrationAgreement.xml:257]: This should be mapped on Observation.note.text

File: resources/zib/zib-AdministrationAgreement.xml [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+    <element id="MedicationDispense.medication[x]:medicationReference">
+      <path value="MedicationDispense.medication[x]" />
+      <sliceName value="medicationReference" />
+      <type>
+        <code value="Reference" />
+        <targetProfile value="http://hl7.org/fhir/StructureDefinition/Medication" />
+        <targetProfile value="http://nictiz.nl/fhir/StructureDefinition/zib-PharmaceuticalProduct" />
+      </type>
+    </element>
```
Comment (Ashwin-nictiz) [inline resources/zib/zib-AdministrationAgreement.xml:203]: Missing NL-CM:9.8.20237 mapping

File: resources/zib/ext-AdministrationAgreement.AdditionalInformation.xml [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    <differential>
+        <element id="Extension.url">
+            <path value="Extension.url" />
+            <fixedUri value="http://nictiz.nl/fhir/StructureDefinition/ext-AdministrationAgreement.AdditionalInformation" />
+        </element>
+        <element id="Extension.value[x]">
+            <path value="Extension.value[x]" />
+            <short value="AdministrationAgreementAdditionalInformation" />
+            <definition value="Additional information includes details on the structure of the agreement made.
+                This element mainly contains information that until now has been structured with ZZ rules. The medication-related topics that are now supported by the ZZ rules are best supported with a system functionality. A process has been started for this by Z-index/KNMP. For now, the following list will be used. This list will be replaced by a thesaurus in the G standard at a later stage." />
```
Comment (Ashwin-nictiz) [inline resources/zib/ext-AdministrationAgreement.AdditionalInformation.xml:44]: To much white space..

File: resources/zib/zib-AdministrationAgreement.xml (hunk 3 above)
Comment (jduwel) [reply resources/zib/zib-AdministrationAgreement.xml:257]: MedicationDispense.note.text. But yes, correct

File: resources/zib/zib-AdministrationAgreement.xml (hunk 2 above)
Comment (jduwel) [reply resources/zib/zib-AdministrationAgreement.xml:174]: This is the only place where we put mapping instructions in .mapping.comment. I think the instructions should be put in .definition.

Re: Ashwin, I don't think further discussion on cardinality is needed. The system just needs to fill .status to the best of its capabilities.

File: resources/zib/zib-AdministrationAgreement.xml [hunk 6]
```diff
@@ -0,0 +1,274 @@
...
+      <path value="MedicationDispense.modifierExtension.value[x]" />
+      <short value="AdministrationAgreementStopType" />
+      <definition value="Stop type, the manner in which this medication is discontinued (temporary or definitive)." />
+      <alias value="ToedieningsafspraakStopType" />
+      <mapping>
+        <identity value="zib-administrationagreement-v1.0.3-2020EN" />
+        <map value="NL-CM:9.8.22498" />
+        <comment value="AdministrationAgreementStopType" />
+      </mapping>
+    </element>
```
Comment (jduwel) [inline resources/zib/zib-AdministrationAgreement.xml:154]: Excel mapping says: "Plus uitleg over gebruik MedicationDispense.statusReason. Dat mag samen met deze extensie maar vervangt deze extensie niet."

Don't see that here (or at .statusReason for that matter)

File: resources/zib/zib-AdministrationAgreement.xml [hunk 7]
```diff
@@ -0,0 +1,274 @@
...
+      <sliceName value="periodOfUsePeriod" />
+      <max value="1" />
+      <type>
+        <code value="Extension" />
+        <profile value="http://nictiz.nl/fhir/StructureDefinition/ext-TimeInterval-Period" />
+      </type>
+    </element>
+    <element id="MedicationDispense.extension:periodOfUsePeriod.value[x]">
+      <path value="MedicationDispense.extension.value[x]" />
+      <short value="PeriodOfUse" />
```
Comment (jduwel) [inline resources/zib/zib-AdministrationAgreement.xml:96]: If you put a short of "PeriodOfUse / Duration" on periodOfUseDuration, shouldn't this be "PeriodOfUse / Period"?

File: resources/zib/zib-AdministrationAgreement.xml (hunk 6 above)
Comment (jduwel) [reply resources/zib/zib-AdministrationAgreement.xml:154]: Nor in ext-StopType itself. ext-StopType talks about .status though. Don't know if that's whats intended

File: resources/zib/zib-AdministrationAgreement.xml (hunk 3 above)
Comment (Ashwin-nictiz) [reply resources/zib/zib-AdministrationAgreement.xml:257]: Typo fix thanks! @jduwel

File: resources/zib/zib-AdministrationAgreement.xml (hunk 2 above)
Comment (jduwel) [reply resources/zib/zib-AdministrationAgreement.xml:174]: > I think the instructions should be put in .definition.
Or on .comment. Not sure actually

File: resources/zib/zib-AdministrationAgreement.xml (hunk 1 above)
Comment (ArdonToonstra) [reply resources/zib/zib-AdministrationAgreement.xml:18]: Thnx. Resloved.

File: resources/zib/zib-AdministrationAgreement.xml (hunk 2 above)
Comment (ArdonToonstra) [reply resources/zib/zib-AdministrationAgreement.xml:174]: Moved from the `mapping.comment` to comment on `status`.

File: resources/zib/zib-AdministrationAgreement.xml (hunk 4 above)
Comment (ArdonToonstra) [reply resources/zib/zib-AdministrationAgreement.xml:203]: Thnx. Fixed.

File: resources/zib/zib-AdministrationAgreement.xml (hunk 7 above)
Comment (ArdonToonstra) [reply resources/zib/zib-AdministrationAgreement.xml:96]: Jub! Thnx

File: resources/zib/zib-AdministrationAgreement.xml (hunk 3 above)
Comment (ArdonToonstra) [reply resources/zib/zib-AdministrationAgreement.xml:274]: Nice! Fixed in MediationProfiles branch.

File: resources/zib/zib-AdministrationAgreement.xml (hunk 6 above)
Comment (ArdonToonstra) [reply resources/zib/zib-AdministrationAgreement.xml:162]: Relates to https://github.com/Nictiz/Nictiz-R4-zib2020/issues/114 and I made a to do for my self to improve this.

**Comment by ArdonToonstra** [comment]
Merded in MedicationProfiles.

**Pull request closed without merging**